File: OnlineQuizJava/app/src/main/java/com/example/onlinequizjava/fragments/HomeFragment.java
package com.example.onlinequizjava.fragments;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.onlinequizjava.MainActivity;
import com.example.onlinequizjava.R;

public class HomeFragment extends Fragment {

    public HomeFragment() {
        super(R.layout.fragment_home);
    }

    /*
        Ana ekran.

        Kullanıcı:
        - Oda oluşturabilir
        - Var olan odaya katılabilir
    */

    @Override
    public void onViewCreated(
            @NonNull View view,
            @Nullable Bundle savedInstanceState
    ) {
        super.onViewCreated(view, savedInstanceState);

        Button btnCreateRoom = view.findViewById(R.id.btnCreateRoom);
        Button btnJoinRoom = view.findViewById(R.id.btnJoinRoom);

        btnCreateRoom.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ((MainActivity) requireActivity()).openCreateRoomFragment();
            }
        });

        btnJoinRoom.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ((MainActivity) requireActivity()).openJoinRoomFragment();
            }
        });
    }
}

File: OnlineQuizKotlin/app/src/main/java/com/example/onlinequizkotlin/fragments/HomeFragment.kt
package com.example.onlinequizkotlin.fragments


import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.fragment.app.Fragment
import com.example.onlinequizkotlin.MainActivity
import com.example.onlinequizkotlin.R

class HomeFragment : Fragment(R.layout.fragment_home) {

    /*
        Ana ekran.

        Kullanıcı burada iki işlemden birini seçer:
        1. Oda oluştur
        2. Odaya giriş yap
    */

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val btnCreateRoom = view.findViewById<Button>(R.id.btnCreateRoom)
        val btnJoinRoom = view.findViewById<Button>(R.id.btnJoinRoom)

        btnCreateRoom.setOnClickListener {
            (requireActivity() as MainActivity).openCreateRoomFragment()
        }

        btnJoinRoom.setOnClickListener {
            (requireActivity() as MainActivity).openJoinRoomFragment()
        }
    }
}